src/subscriber.ts

```diff
@@ -57,67 +51,85 @@ export class Subscriber {
         );
     }
 
-    private _initTransactions = (): void => {
-      this._initializedTransactions = {};
-      for (const subscription of this.subscribe.transactions) {
-          this._initializedTransactions[subscription.name] = false;
-      }
-    }
-
     private  _triggerDebugActions = async (): Promise<void> => {
       this.logger.debug('debug mode active')
     }
 
-    private _subscribeTransactions = async (): Promise<void> =>{
-      const freeBalance: FreeBalance = {};
-      await asyncForEach(this.subscribe.transactions, async (account) => {
-          const { data: { free: previousFree } } = await this.api.query.system.account(account.address);
-          freeBalance[account.address] = previousFree;
-          
-          await this.api.query.system.account(account.address, async (accountInfo) => { this._transactionHandler(accountInfo,account,freeBalance) });
-      });
-  }
+    private _handleNewHeadSubscriptions = async (): Promise<void> =>{
 
-  private _transactionHandler = async (accountInfo: AccountInfo, account: Subscribable, freeBalance: FreeBalance): Promise<void> => {
-    this.logger.info(`The nonce for ${account.name} is ${accountInfo.nonce}`);
+      this.api.rpc.chain.subscribeFinalizedHeads(async (header) => {
+       
+        await this._blocksHandler(header)
+      })
+    }
 
-    if (this._initializedTransactions[account.name]) {
-        const data: TransactionData = {
-            name: account.name,
-            address: account.address,
-            networkId: this.networkId
-            //add hash
-        };
+    private _blocksHandler = async (header: Header): Promise<void> =>{
 
-        const currentFreeBalance = accountInfo.data.free;
+      const hash = header.hash.toHex()
+      const block = await this.api.rpc.chain.getBlock(hash)
+      this.logger.debug(`block:`)
+      this.logger.debug(JSON.stringify(block))
 
-        this._setTransactionType(data,currentFreeBalance,freeBalance[account.address],account)
+      block.block.extrinsics.forEach( async (extrinsic) => {
 
-        freeBalance[account.address] = currentFreeBalance;
+        this._transferBalancesExtrinsicHandler(extrinsic, hash)
+  
+      })
 
-        try {
-            await this.notifier.newTransaction(data);
-        } catch (e) {
-            this.logger.error(`could not notify transaction: ${e.message}`);
-        }
-    } else {
-        this._initializedTransactions[account.name] = true;
     }
-  }
 
-  private _setTransactionType = (data: TransactionData, currentFreeBalance: Balance, previousFreeBalance: Balance, account: Subscribable): void => {
-    
-    // check if the action was performed by the account or externally
-    const change = currentFreeBalance.sub(previousFreeBalance);
-    if (!change.gt(ZeroBalance)) {
-        this.logger.info(`Action performed from account ${account.name}`);
+    private _transferBalancesExtrinsicHandler = async (extrinsic: Extrinsic, blockHash: string): Promise<void> =>{
```

Comment:
It would be very nice to write a small unit test for this, maybe injecting a notifier stub, is that possible?

Reply:
Done, thanks for the hint